clusters/build-clusters/01_cluster/hack/generate-bw-items.sh

```diff
@@ -14,31 +14,48 @@ readonly WORKDIR
 CLUSTER_NAME="build01"
 readonly CLUSTER_NAME
 
+SAVED_CURRENT_CONTEXT="$(oc config current-context)"
+readonly SAVED_CURRENT_CONTEXT
+
+function recover_current_context {
+  echo "recovering the current context"
+  kubectl config use-context "${SAVED_CURRENT_CONTEXT}"
+}
+
+trap recover_current_context EXIT
 kubectl config use-context $CLUSTER_NAME
 
+SED_COMMAND="sed"
+if [[ "$(uname -s)" == "Darwin" ]]; then
+  SED_COMMAND="gsed"
+fi
+
 generate_kubeconfig() {
   local sa
   sa=$1
   local config
-  config="${WORKDIR}/sa.${sa}.${CLUSTER_NAME}.config"
-  oc --kubeconfig "${KUBECONFIG_BUILD01}" sa create-kubeconfig -n ci "${sa}" > "${config}"
-  # oc config rename-context is not enough, as we then end up
-  # with multiple users with the same name when they get merged
-  sed -i "s/${sa}/${CLUSTER_NAME}/g" $config
+  config=$2
+  oc sa create-kubeconfig -n ci "${sa}" > "${config}"
+  "${SED_COMMAND}" -i "s/${sa}/${CLUSTER_NAME}/g" "${config}"
 }
 
 declare -a SAArray=( "config-updater" "deck" "plank" "sinker" "hook" "ca-cert-issuer" "crier" )
 
 # Iterate the string array using for loop
 for name in ${SAArray[@]}; do
-   generate_kubeconfig ${name}
+  sa_config="${WORKDIR}/sa.${name}.${CLUSTER_NAME}.config"
+  generate_kubeconfig "${name}" "${sa_config}"
```

Comment:
I would make that part of the `generate_kubeconfig` function so if someone generates a kubeconfig outside of this loop, it will get validated so well

Reply:
Done.